### DISCUSSION TITLE
Get the framebuffer as a raw 2D array

### DISCUSSION THREAD

**Discussion by ghost** [parasyte/pixels#296] [Q&A]
Hi, 
I am using pixels for some time now and i would like to know a way to access the framebuffer like a 2D array (width, height), however I am not able to find something. i need this so i can change a pixel's color in a specified position, eg:
```rust
let (x, y) = (32, 43);
let color: [u8; 4] = [0xF2, 0xA3, 0x4D, 0xFF]
let pixels = /* Creating the buffer etc */
pixels[x][y] = color;
```

**Comment by parasyte** [comment]
Honestly there is no need for a 2D array, but you might be able to cast the slice to a slice-of-slices with something like [`bytemuck`](https://crates.io/crates/bytemuck) if you really want to.

Traditionally, pixel plotting would be done with the formula:

```rust
let index = (y * width + x) * 4;
```

Since the color is itself a slice, you can use `slice::copy_from_slice()` to copy its contents to the buffer:

```rust
buffer[index..index + 4].copy_from_slice(&color);
```

Something like this.

Have a look at the minimal example to see how it copies colors to the buffer: https://github.com/parasyte/pixels/blob/4ee50064816f7a32b3dc81f9dfe33cb66b292947/examples/minimal-winit/src/main.rs#L103-L124

**Reply by ghost** [reply]
Just a question, `buffer` is the `Pixels` framebuffer or something else?

**Reply by parasyte** [reply]
Yes, that would be the framebuffer from `Pixels::get_frame()`.
